File: Visual Basic .NET/TreeGridflowView/ConfigurationData.vb
'
' * Xceed DataGrid for WPF - SAMPLE CODE - TreeGridflowView View Sample Application
' * Copyright (c) 2007-2024 Xceed Software Inc.
' * 
' * [ConfigurationData.cs]
' * 
' * This file is part of the Xceed DataGrid for WPF product. The use 
' * and distribution of this Sample Code is subject to the terms 
' * and conditions referring to "Sample Code" that are specified in 
' * the XCEED SOFTWARE LICENSE AGREEMENT accompanying this product.
' *
' 


Imports Microsoft.VisualBasic
Imports System.ComponentModel
Imports System.Windows

Namespace Xceed.Wpf.DataGrid.Samples.TreeGridflowView
  Public Class ConfigurationData
	  Implements INotifyPropertyChanged
	#Region "Singleton Property"

	Public Shared ReadOnly Singleton As New ConfigurationData()

	#End Region

	#Region "CONSTRUCTORS"

	Private Sub New()
	End Sub

	#End Region

	#Region "FlowDirection Property"

	Public Property FlowDirection() As FlowDirection
	  Get
		Return m_flowDirection
	  End Get

	  Set(ByVal value As FlowDirection)
		If value <> m_flowDirection Then
		  m_flowDirection = value
		  Me.RaisePropertyChanged("FlowDirection")
		  Me.RaisePropertyChanged("FlowDirectionLeftToRight")
		  Me.RaisePropertyChanged("FlowDirectionRightToLeft")
		End If
	  End Set
	End Property

	Public Property FlowDirectionLeftToRight() As Boolean
	  Get
		Return (Me.FlowDirection = FlowDirection.LeftToRight)
	  End Get

	  Set(ByVal value As Boolean)
		If value Then
		  Me.FlowDirection = FlowDirection.LeftToRight
		End If
	  End Set
	End Property

	Public Property FlowDirectionRightToLeft() As Boolean
	  Get
		Return (Me.FlowDirection = FlowDirection.RightToLeft)
	  End Get

	  Set(ByVal value As Boolean)
		If value Then
		  Me.FlowDirection = FlowDirection.RightToLeft
		End If
	  End Set
	End Property

	Private m_flowDirection As FlowDirection = FlowDirection.LeftToRight

	#End Region

	#Region "ShowColumnManagerRow Property"

	Public Property ShowColumnManagerRow() As Boolean
	  Get
		Return m_showColumnManagerRow
	  End Get

	  Set(ByVal value As Boolean)
		If value <> m_showColumnManagerRow Then
		  m_showColumnManagerRow = value
		  Me.RaisePropertyChanged("ShowColumnManagerRow")
		End If
	  End Set
	End Property

	Private m_showColumnManagerRow As Boolean = True

	#End Region

  #Region "AllowFixedColumnSplitterDrag Property"

	Public Property AllowFixedColumnSplitterDrag() As Boolean
	  Get
		Return m_allowFixedColumnSplitterDrag
	  End Get

	  Set(ByVal value As Boolean)
		If value <> m_allowFixedColumnSplitterDrag Then
		  m_allowFixedColumnSplitterDrag = value
		  Me.RaisePropertyChanged("AllowFixedColumnSplitterDrag")
		End If
	  End Set
	End Property

	Private m_allowFixedColumnSplitterDrag As Boolean = True

	#End Region

	#Region "ShowFixedColumnSplitter Property"

	Public Property ShowFixedColumnSplitter() As Boolean
	  Get
		Return m_showFixedColumnSplitter
	  End Get

	  Set(ByVal value As Boolean)
		If value <> m_showFixedColumnSplitter Then
		  m_showFixedColumnSplitter = value
		  Me.RaisePropertyChanged("ShowFixedColumnSplitter")
		End If
	  End Set
	End Property

	Private m_showFixedColumnSplitter As Boolean = True

	#End Region

	#Region "INotifyPropertyChanged Implementation"

	Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

	Protected Sub RaisePropertyChanged(ByVal name As String)
	  If Me.PropertyChangedEvent IsNot Nothing Then
		RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(name))
	  End If
	End Sub

	#End Region
  End Class
End Namespace

File: CSharp/TreeGridflowView/ConfigurationData.cs
/*
 * Xceed DataGrid for WPF - SAMPLE CODE - TreeGridflowView View Sample Application
 * Copyright (c) 2007-2024 Xceed Software Inc.
 * 
 * [ConfigurationData.cs]
 * 
 * This file is part of the Xceed DataGrid for WPF product. The use 
 * and distribution of this Sample Code is subject to the terms 
 * and conditions referring to "Sample Code" that are specified in 
 * the XCEED SOFTWARE LICENSE AGREEMENT accompanying this product.
 *
 */

using System.ComponentModel;
using System.Windows;

namespace Xceed.Wpf.DataGrid.Samples.TreeGridflowView
{
  public class ConfigurationData : INotifyPropertyChanged
  {
    #region Singleton Property

    public static readonly ConfigurationData Singleton = new ConfigurationData();

    #endregion

    #region CONSTRUCTORS

    private ConfigurationData()
    {
    }

    #endregion

    #region FlowDirection Property

    public FlowDirection FlowDirection
    {
      get
      {
        return m_flowDirection;
      }

      set
      {
        if( value != m_flowDirection )
        {
          m_flowDirection = value;
          this.RaisePropertyChanged( "FlowDirection" );
          this.RaisePropertyChanged( "FlowDirectionLeftToRight" );
          this.RaisePropertyChanged( "FlowDirectionRightToLeft" );
        }
      }
    }

    public bool FlowDirectionLeftToRight
    {
      get
      {
        return ( this.FlowDirection == FlowDirection.LeftToRight );
      }

      set
      {
        if( value )
        {
          this.FlowDirection = FlowDirection.LeftToRight;
        }
      }
    }

    public bool FlowDirectionRightToLeft
    {
      get
      {
        return ( this.FlowDirection == FlowDirection.RightToLeft );
      }

      set
      {
        if( value )
        {
          this.FlowDirection = FlowDirection.RightToLeft;
        }
      }
    }

    private FlowDirection m_flowDirection = FlowDirection.LeftToRight;

    #endregion

    #region AllowFixedColumnSplitterDrag Property

    public bool AllowFixedColumnSplitterDrag
    {
      get
      {
        return m_allowFixedColumnSplitterDrag;
      }

      set
      {
        if( value != m_allowFixedColumnSplitterDrag )
        {
          m_allowFixedColumnSplitterDrag = value;
          this.RaisePropertyChanged( "AllowFixedColumnSplitterDrag" );
        }
      }
    }

    private bool m_allowFixedColumnSplitterDrag = true;

    #endregion AllowFixedColumnSplitterDrag Property

    #region ShowColumnManagerRow Property

    public bool ShowColumnManagerRow
    {
      get
      {
        return m_showColumnManagerRow;
      }

      set
      {
        if( value != m_showColumnManagerRow )
        {
          m_showColumnManagerRow = value;
          this.RaisePropertyChanged( "ShowColumnManagerRow" );
        }
      }
    }

    private bool m_showColumnManagerRow = true;

    #endregion

    #region ShowFixedColumnSplitter Property

    public bool ShowFixedColumnSplitter
    {
      get
      {
        return m_showFixedColumnSplitter;
      }

      set
      {
        if( value != m_showFixedColumnSplitter )
        {
          m_showFixedColumnSplitter = value;
          this.RaisePropertyChanged( "ShowFixedColumnSplitter" );
        }
      }
    }

    private bool m_showFixedColumnSplitter = true;

    #endregion

    #region INotifyPropertyChanged Implementation

    public event PropertyChangedEventHandler PropertyChanged;

    protected void RaisePropertyChanged( string name )
    {
      if( this.PropertyChanged != null )
      {
        this.PropertyChanged( this, new PropertyChangedEventArgs( name ) );
      }
    }

    #endregion
  }
}
